src/app/dock_widgets/process/gt_processdock.cpp

```diff
@@ -1773,7 +1773,7 @@ GtProcessDock::deleteProcessElements(const QList<QModelIndex>& indexList)
         {
             gtDataModel->reduceToParents(objects);
 
-            auto cmd = gtApp->makeCommand(gtApp->currentProject(),
+            auto cmd = gtApp->makeCommand(m_taskGroup,
```

Comment:
Consider this: You have many many tasks, and somewhere deeply nested you delete a process component. The code still check all other tasks, you could however use the the `find_lowest_ancestor` function to find the "commonParent". It may be the task group or it may be the an object that has a lot less children, thus could perform better still.
```suggestion
            auto* commonParent = gt::find_lowest_ancestor(objects, gt::get_parent_object);
            assert(commonParent);
            auto cmd = gtApp->makeCommand(commonParent,
```